docs/zkapps/o1js/basic-concepts.md

```diff
@@ -27,109 +27,125 @@ zkApp programmability is not yet available on the Mina Mainnet. You can get star
+const x = Field(-1); // If you try printing this to the screen, you will see a very big integer
+```
 
-PrivateKey, PublicKey, Signature; // useful for accounts and signing
-new Group(x, y); // a point on our elliptic curve, accepts two Fields/numbers/strings
-Scalar; // the corresponding scalar field (different than Field)
+## Methods in the Field Type
 
-CircuitString.from('some string'); // string of max length 128
-```
+`Field` type includes a lot of built in methods. You can call all methods in the `Field` type either with a `Field` or with a "field-like" value. In the latter case, the argument is implicity converted to a `Field`.
 
-In the case of `Field` and `Bool`, you can also call the constructor without `new`:
+All common arithmetic operations exist in the `Field` type. They create and return a new `Field` element.
 
+<!-- prettier-ignore -->
 ```ts
-let x = Field(10);
-let b = Bool(true);
+const x = Field(45);
+const y = Field(78);
+
+const sum = x.add(y);
+const sub = x.sub(y);
+const mul = x.mul(47); // You can also use "field-like" values
+const div = x.div(89); // This is a modular division
+const neg = x.neg(); // This is equivalent to x.mul(-1)
+
+const square = x.square(); // x^2, equivalent to x.mul(x)
+const sqrt = x.sqrt(); // Modular square root of x
+const inv = x.inv(); // Modular inverse of x
 ```
 
-## Conditionals
-
-Traditional conditional statements are not supported by o1js:
+You can also perform logical operations and comparisons with `Field` elements. These methods return a [Bool](./common-types-and-functions.md#bool) element, which is another common o1js built in type, equivalent to `boolean` in other languages and frameworks.
 
+<!-- prettier-ignore -->
 ```ts
-// this will NOT work
-if (foo) {
-  x.assertEquals(y);
-}
+const x = Field(45);
+
+x.equals(45); // True
+x.greaterThan(45); // False
+x.greaterThanOrEquals(45); // True
+x.lessThan(45); // False
+x.lessThanOrEquals(45); // True
+
+// It is also possible to check if a Field is even or not
+x.isEven(); // False
 ```
 
-Instead, use the o1js built-in `Circuit.if()` method, which is a ternary operator:
+## Assertions in o1js
 
+If you want to reject a TX (transaction) in your zkApp smart contract, you can use assertions in o1js. If you create an assertion inside your ZKP, the proof fails if the assertion does not evaluate to `true`. All conditional methods on the `Field` type can be used as assertions.
+
+<!-- prettier-ignore -->
 ```ts
-const x = Circuit.if(new Bool(foo), a, b); // behaves like `foo ? a : b`
+const x = Field(45);
+
+x.assertEquals(45); // True - passes
+x.assertGreaterThan(45); // False - throws
+x.assertGreaterThanOrEquals(45); // True - passes
+x.assertLessThan(45); // False - throws
+x.assertLessThanOrEquals(45); // True - passes
 ```
 
-## Functions
+You can think assertions in o1js similar to `require()` function in Solidity.
 
-Functions work as you would expect in TypeScript. For example:
+## Printing a Field to the Screen
```

Comment:
I'm not a fan of this section at all, because toString etc don't work inside a circuit. Why not teach about Provable.log() here?

Reply:
Also teaching about constants vs variables would be nice